### PR TITLE
#167190398 Implement CRUD operations for articles

### PR DESCRIPTION
**Pull request by chuxmykel**
#### What does this PR do?
- This PR implements CRUD operations on the platform.

#### Description of Task to be completed?
- Create CRUD pages and implement the functionality
- Create a `404` page to give feedback to users on searching for non-existent articles

#### How should this be manually tested?
- visit the staging link that will be created from this PR and try the following CRUD operations
  - Create article `/create-article`
  - Read article `/articles/<slug>`
  - Update article `/articles/<slug>/edit`

#### Any background context you want to provide?
- Some work will still be done on the CSS. I need to sync with @cvjude to brainstorm on where to position the delete button.
- Work is done and E2E is in progress...
  - [Create article e2e video](https://drive.google.com/file/d/1iLCAQhS05KFqGqE0IOSx8LTpwO3CrTto/view?usp=sharing)
  - [Read article e2e video](https://drive.google.com/file/d/1LTyfoWoOdgJwCyOUndXLdC4m5BFGBYTL/view?usp=sharing)
  - [Update article e2e video](https://drive.google.com/file/d/1KuLrdYfTeMucoRzc-cKHQTVvxksNR-4o/view?usp=sharing)
  - [Delete article e2e video](https://drive.google.com/file/d/1oyfMuYgWtRPeQUdgvcf4WN0oEVBP1Lut/view?usp=sharing)
  - [404 e2e video](https://drive.google.com/file/d/1NapP7J4jwBljL0wNthAEwo6LAwtrnRZN/view?usp=sharing)

#### What are the relevant pivotal tracker stories?
- [#167190398](https://www.pivotaltracker.com/story/show/167190398)

### CODE REVIEW

**Review by obayomi96 — COMMENTED** [review]
Good job on pulling this off @chuxmykel , this is a job well done and I commend you on the use of editor.js framework as well. However, I noticed you did not add an end-to-end test for these features. Can you kindly do that @chuxmykel . 

Again great work.

**Review by encodedBicoding — APPROVED** [review]
LVGTM! Just fix the merge conflicts.

**Comment by chuxmykel** [comment]
> Good job on pulling this off @chuxmykel , this is a job well done and I commend you on the use of editor.js framework as well. However, I noticed you did not add an end-to-end test for these features. Can you kindly do that @chuxmykel .
> 
> Again great work.

Sorry about that. I just got home and I'm currently on it. I'll add the link to the PR as soon as I'm done.

**Comment by chuxmykel** [comment]
> LVGTM! Just fix the merge conflicts.

Thanks. I'm working on getting that done now.

**Review by Lundii — APPROVED** [review]
LGTM

**Review by Lundii — COMMENTED** [review]
Nice work with the task. I want to ask will it be possible to reduce the width of the featured image of the article so all images can display with a good proportion.

File: src/views/Articles/ReadArticle/ReadArticle.js [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+      <>
+        {loading ? loader : (
+          <div className="read-article">
+            <div className="article-header vertical-center">
+              <h1 className="title">{title}</h1>
+              <p className="description">{description}</p>
+              <div className="article-user center">
+                <div className="article-user-details">
+                  <div className="article-img-div center">
+                    <img
```
Comment (cvjude) [inline src/views/Articles/ReadArticle/ReadArticle.js:116]: Nice work Chux, I think you should load a custom placeHolder image if no image data comes from the database.
Here is the one we've been using 'https://res.cloudinary.com/drdje1skj/image/upload/v1567427029/profile-placeholder_gvxkia.gif'

**Review by cvjude — APPROVED** [review]
LGTM

File: src/views/Articles/ReadArticle/ReadArticle.js (hunk 1 above)
Comment (chuxmykel) [reply src/views/Articles/ReadArticle/ReadArticle.js:116]: All right. I will do that now. Thanks

**Comment by chuxmykel** [comment]
> Nice work with the task. I want to ask will it be possible to reduce the width of the featured image of the article so all images can display with a good proportion.

That is possible. But that will depend entirely on the PO

**Comment by Lundii** [comment]
> > Nice work with the task. I want to ask will it be possible to reduce the width of the featured image of the article so all images can display with a good proportion.
> 
> That is possible. But that will depend entirely on the PO

Alright.

**Review by Hadeneekeh — APPROVED** [review]
LGTM!!!

**Review by chiboycalix — APPROVED** [review]
Nice work chuks. This Looks good to me

**Review by obayomi96 — APPROVED** [review]
Looks good to me

**Review by topseySuave — APPROVED** [review]

**Pull request merged**